app/models/step_run.rb

```diff
@@ -138,6 +138,15 @@ class StepRun < ApplicationRecord
   delegate :download_url, to: :build_artifact
   scope :not_failed, -> { where.not(status: [:ci_workflow_failed, :ci_workflow_halted, :build_not_found_in_store, :build_unavailable, :deployment_failed]) }
 
+  def self.runs_between(start_step_run, end_step_run)
+    return none if start_step_run.nil? && end_step_run.nil?
+    return where(id: end_step_run.id) if start_step_run.nil?
+
+    where(step_id: (start_step_run || end_step_run).step_id)
+      .where("scheduled_at BETWEEN ? AND ?", start_step_run.scheduled_at, end_step_run&.scheduled_at)
+      .where.not(id: start_step_run.id)
```

Comment:
we can do `scheduled_at > start_step_run.scheduled_at` and avoid the last `where.not`.